docs/markdoc-component-documentation.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Authoring Component Examples
+description: This page shows off what each of the components we're using in our docs looks like.
+---
+
+This page contains examples of all of the native Markdoc node and the custom markdoc tags that we are using in our docs.
+
+To see the rendered version of these tags, move this file into the `next/pages/getting-started` directory, run the local server for the site, and navigate to the `getting-started/markdoc-component-documentation` page in your browser.
+
+# Header 1
+
+## Header 2
+
+### Header 3
+
+#### Header 4
+
+##### Header 5
+
+## Regular text
+
+When the wire was despatched he had a cup of tea; over it he told me of a diary kept by Jonathan Harker when abroad, and gave me a typewritten copy of it, as also of Mrs. Harker's diary at Whitby. "Take these," he said, "and study them well. When I have returned you will be master of all the facts, and we can then better enter on our inquisition. Keep them safe, for there is in them much of treasure. You will need all your faith, even you who have had such an experience as that of to-day. What is here told," he laid his hand heavily and gravely on the packet of papers as he spoke, "may be the beginning of the end to you and me and many another; or it may sound the knell of the Un-Dead who walk the earth. Read all, I pray you, with the open mind; and if you can add in any way to the story here told do so, for it is all-important. You have kept diary of all these so strange things; is it not so? Yes! Then we shall go through all these together when we meet." He then made ready for his departure, and shortly after drove off to Liverpool Street. I took my way to Paddington, where I arrived about fifteen minutes before the train came in.
+
+## Complex Header Where You Definitely Want to Use an Anchor Link {% #complex-header %} 
+
+```
+## Complex Header Where You Definitely Want to Use an 
+Anchor Link {\% #complex-header %} 
+```
+
+# Vanilla Markdown Nodes
+
+**Bold**
+
+_Italic_
+
+[Links](/docs/nodes)
+
+Vanilla markdown images work, but we should really be using the image tag because it optimizes the images, makes them more accessible, and handles resizing them for us.
+![Images](/images/concepts/assets/asset-activity-observation.png)
+
+Lists
+- Item 1
+- Item 1
+- Item 1
+
+> Quotes
+
+`Inline code`
+
+```
+Code fences
+```
+
+Markdoc doesn't have syntax highlighting out of the box. We're going to need to add that in. When it get's added in, this python code should be beautifully highlighted.
+
+```python
```

Comment:
The one thing that's stopped me from migrating content to Markdoc is that we're not currently able to pull in code snippets from the examples folder (`/examples`).

The way we currently do this has two steps:

1. Use a `file` parameter after the language specified on a code fence, ex:
   ````
   ```python file=/docs_snippets/docs_snippets/concepts/assets/asset_group.py
   ```
   ````
2. Run `make mdx-format`, which injects the contents of the specified file into the code block. In this example, we'd then see: 
   ````
   ```python file=/docs_snippets/docs_snippets/concepts/assets/asset_group.py
   from dagster import asset


   @asset
   def upstream_asset():
       return [1, 2, 3]


   @asset
   def downstream_asset(upstream_asset):
       return upstream_asset + [4]
   ```
   ````

I'm not sure of the right/easy way to handle this in Markdoc, but if I had this, I could start migrating everything over and using these fabulous components. Perhaps discuss and then a follow-up PR to this one?

Reply:
I think we should discuss. I want to understand more about why we are doing code samples this way because there might be an easier approach.